### ISSUE TITLE
tf.nn.atrous_conv2d crashes(aborts) when rate is large value

### ISSUE THREAD

**Issue by DNXie** [tensorflow/tensorflow#46915]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 18.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below):2.1.0
- Python version:3.7.6
- Bazel version (if compiling from source):N/A
- GCC/Compiler version (if compiling from source):N/A
- CUDA/cuDNN version:N/A
- GPU model and memory:N/A

**Describe the current behavior**
`tf.nn.atrous_conv2d` crashes(aborts) when `rate` is large value

**Describe the expected behavior**
expect an exception message if the input unexpected instead of crash. 

**Standalone code to reproduce the issue**
~~~python
import tensorflow as tf
import numpy as np
tf.nn.atrous_conv2d(value=np.ones((1,1,1,5)), filters=np.ones((1,1,5,1)), rate=2147483647, padding='SAME')
~~~

~~~python
2021-02-04 04:47:25.891213: F tensorflow/core/framework/tensor_shape.cc:353] Check failed: 0 <= new_num_elements (0 vs. -1)
Aborted (core dumped)
~~~

**Comment by ravikyram** [comment]
I have tried in colab with TF versions 2.1, 2.4, nightly versions (`2.5.0-dev20210203`) and was able to reproduce the issue. Please, find the gist [here](https://colab.research.google.com/gist/ravikyram/1fda44be7360e4382bedf8f9c6c8dfb9/untitled658.ipynb). Thanks!

**Comment by DNXie** [comment]
BTW, `tf.nn.atrous_conv2d_transpose` has similar crash which is due to large `rate`.

~~~python
import tensorflow as tf
import numpy as np
tf.nn.atrous_conv2d_transpose(value=np.ones((10,1,1,1)), filters=np.ones((1,1,1,1)), rate=1356819205, padding='SAME', output_shape=[1,1,1,1])
~~~

Error Message:
~~~python
2021-04-15 00:08:19.741409: F tensorflow/core/framework/tensor_shape.cc:397] Check failed: size >= 0 (-37160523141231366 vs. 0)
Aborted (core dumped)
~~~

**Comment by saikumarchalla** [comment]
Colab crashes in TF Nightly 2.6 as well.Please find the gist [here](https://colab.research.google.com/gist/saikumarchalla/b44477be4d8ec2fdc1028f4684ec4cdf/untitled92.ipynb).Thanks!

**Comment by yongtang** [comment]
Added a PR #57854 for the fix.
